pkg/media/ivf-writer_test.go

```diff
@@ -0,0 +1,151 @@
+			err:          fmt.Errorf("Payload is not large enough to container header"),
+			closeErr:     nil,
+		},
+		{
+			fileName:     "/tmp/pions_testIVFPacket3.ivf",
+			message:      "IVFWriter shouldn't be able to write something else than an IVF packet",
+			messageClose: "IVFWriter should be able to close the file",
+			packet: &rtp.Packet{
+				Header: rtp.Header{
+					Marker:           true,
+					Extension:        true,
+					ExtensionProfile: 1,
+					ExtensionPayload: []byte{0xFF, 0xFF, 0xFF, 0xFF},
+					Version:          2,
+					PayloadOffset:    20,
+					PayloadType:      defaultPayloadTypeOpus,
+					SequenceNumber:   27023,
+					Timestamp:        3653407706,
+					SSRC:             476325762,
+					CSRC:             []uint32{},
+				},
+				Payload: rawPkt[20:],
+				Raw:     rawPkt,
+			},
+			err:      nil, // TODO: Update expected value
+			closeErr: nil,
+		},
+		{
+			fileName:     "/tmp/pions_testIVFPacket4.ivf",
+			message:      "IVFWriter should be able to write an IVF packet",
+			messageClose: "IVFWriter should be able to close the file",
+			// TODO: Is this a valid VP8 packet ?
+			packet: &rtp.Packet{
+				Header: rtp.Header{
+					Marker:           true,
+					Extension:        true,
+					ExtensionProfile: 1,
+					ExtensionPayload: []byte{0xFF, 0xFF, 0xFF, 0xFF},
+					Version:          2,
+					PayloadOffset:    20,
+					PayloadType:      defaultPayloadTypeVP8,
+					SequenceNumber:   27023,
+					Timestamp:        3653407706,
+					SSRC:             476325762,
+					CSRC:             []uint32{},
+				},
+				Payload: rawPkt[20:],
+				Raw:     rawPkt,
+			},
+			err:      nil,
+			closeErr: nil,
+		},
+	}
+
+	// First test case has a 'nil' file descriptor
+	writer, err := NewIVFWriter(addPacketTestCase[0].fileName)
+	assert.Nil(err, "IVFWriter should be created")
+	assert.NotNil(writer, "Writer shouldn't be nil")
+	err = writer.fd.Close()
+	assert.Nil(err, "IVFWriter should be able to close the file descriptor")
+	writer.fd = nil
+	addPacketTestCase[0].writer = writer
+
+	// Second test writes tries to write an empty packet
+	writer, err = NewIVFWriter(addPacketTestCase[1].fileName)
+	assert.Nil(err, "IVFWriter should be created")
+	assert.NotNil(writer, "Writer shouldn't be nil")
+	addPacketTestCase[1].writer = writer
+
+	// Third test writes tries to write a Opus packet
+	writer, err = NewIVFWriter(addPacketTestCase[2].fileName)
+	assert.Nil(err, "IVFWriter should be created")
+	assert.NotNil(writer, "Writer shouldn't be nil")
+	addPacketTestCase[2].writer = writer
+
+	// Fourth test writes tries to write a valid VP8 packet
+	writer, err = NewIVFWriter(addPacketTestCase[3].fileName)
+	assert.Nil(err, "IVFWriter should be created")
+	assert.NotNil(writer, "Writer shouldn't be nil")
+	addPacketTestCase[3].writer = writer
+
+	for _, t := range addPacketTestCase {
+		res := t.writer.AddPacket(t.packet)
+		assert.Equal(res, t.err, t.message)
+	}
+
+	for _, t := range addPacketTestCase {
```

Comment:
Maybe I should do that in a single loop ?

Reply:
IMO this is good! it is testing code, so don't worry about making it terse as long as it is manageable :)